publisher/internal/analytics/ethereum/types.go

```diff
@@ -1,11 +1,27 @@
 package ethereum
 
 import (
+	"encoding/json"
 	"fmt"
+	"log"
+	"reflect"
+	"strings"
+	"time"
 
+	"github.com/SyntropyNet/swapscope/publisher/pkg/analytics"
 	"github.com/SyntropyNet/swapscope/publisher/pkg/repository"
+	"github.com/SyntropyNet/swapscope/publisher/pkg/types"
 )
 
+type Operation interface {
+	// Common methods shared by Addition and Removal
+	ExtractFromEventLogs(EventLog, *Analytics) (Operation, error)
```

Comment:
No need to specify `FromEventLogs` since `EventLog` is already a parameter.

Also, it should not depend on `Analytics`. If it depends on some functionality from `Analytics` - it can be imported as an interface at the instantiation of `Operation` implementation(proper term: Dependency injection). The idea here is that the interface methods should be as self sufficient as possible.